### PR TITLE
Renamed `getHeaders()` to `getHeadings()`

### PR DESCRIPTION
**Pull request by natemoo-re**
#### What kind of changes does this PR include?

- New or updated content
- Changes to the docs site code

#### Description

This PR deprecates `getHeaders()` documentation in favor of `getHeadings()`. According to the HTML spec, these are headings. The only part of the spec where the word "headers" appear is in the context of <header> tag. MDN also uses headings to refer the tags.

Depends on https://github.com/withastro/astro/pull/4031

- What does this PR change? Give us a brief description.
- Did you change something visual? A before/after screenshot can be helpful.

### CODE REVIEW

**Comment by natemoo-re** [comment]
Whoops, looks like there are some merge conflicts still. I'll take a stab at resolving these soon.

File: src/pages/en/guides/markdown-content.md [hunk 1]
```diff
@@ -261,9 +261,15 @@ The absolute path of this file (e.g. `/home/user/projects/.../file.md`).
...
 
+#### `getHeadings()`
+
+An async function that returns the headings of the Markdown file. This function returns the type: `{ depth: number; slug: string; text: string }[]`.
+
 #### `getHeaders()`
 
-An async function that returns the headers of the Markdown file. The response follows this type: `{ depth: number; slug: string; text: string }[]`.
+:::caution[Deprecated]
+The `getHeaders()` function has been renamed to `getHeadings()`. Please update your code to use `getHeadings()`.
```
Comment (sarah11918) [inline src/pages/en/guides/markdown-content.md:271]:
```suggestion
The `getHeaders()` function has been renamed. To update an older project, please use `getHeadings()` instead.
```
Just a suggestion to be more explicit about which one to use. Similarly, a note about this should end up in the migration guide. (You could probably use this as-is there.)

**Comment by sarah11918** [comment]
Adding a note that `getHeaders()` also appears on https://docs.astro.build/en/guides/integrations-guide/mdx/#exported-properties which will need to be updated here: https://github.com/withastro/astro/blob/main/packages/integrations/mdx/README.md

**Comment by sarah11918** [comment]
@natemoo-re I was easily able to resolve the conflict here, so I did, FYI.

**Comment by delucis** [comment]
A lot of these changes to site code have already been handled, so I’m going to close this and raise a new PR with just the docs content changes.

**Pull request closed without merging**